JavaIntegration/src/main/java/org/openzen/zencode/java/module/JavaRuntimeMethod.java

```diff
@@ -70,12 +70,30 @@ else if (method.getDeclaringClass().isInterface())
 		this.class_ = class_;
 		this.target = target;
 		this.method = new JavaNativeMethod(class_.javaClass, kind, method.getName(), compile, org.objectweb.asm.Type.getMethodDescriptor(method), method
-				.getModifiers(), header.getReturnType().isGeneric());
+				.getModifiers(), header.getReturnType().isGeneric(), calculateTypeParameterArguments(method, expansion));
 		modifiers = getMethodModifiers(method, implicit, expansion);
 		this.id = id;
 		this.header = header;
 	}
 
+	private static boolean[] calculateTypeParameterArguments(Method method, boolean expansion) {
+		boolean[] typeArguments = new boolean[method.getTypeParameters().length];
```

Comment:
Works for:
- `@StaticExpansionMethod` with `<T>`
- Expansion `@Method` with `<T>`

Not tested: Expansion that itself is generic a generic type (since not yet in ZC)
The equivalent of 
```tsx
public expand <T> T {
    public <U> orElse(other: U): Either<T, U> => LEFT(this);
}
```